Mathlib/Algebra/FreeMonoid/Symbols.lean

```diff
@@ -0,0 +1,40 @@
+/-
+Copyright (c) 2024 Hannah Fechtner. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hannah Fechtner
+-/
+
+import Mathlib.Algebra.FreeMonoid.Basic
+import Mathlib.Data.Finset.Basic
+
+/-!
+# The finite set of symbols in a FreeMonoid element
+
+This is separated from the main FreeMonoid file, as it imports the finiteness hierarchy
```

Comment:
I still don't quite buy that argument, @jcommelin. The only extra thing it imports is `Finset`. If that's too much, then we should be working on it rather than artificially splitting later files